src/Database/TypeInterface.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3.0
```

Comment:
I thought that, but then I realised that it wouldn't actually cause any BC breaks, as custom type classes will have implemented these types already. I'm happy to either change the version number, or re-submit the pull request, either is fine depending on where the core team feel that merging change would be most helpful.